### DISCUSSION TITLE
OverlayPanel: prop to specify placement relative to target

### DISCUSSION THREAD

**Discussion by jacobg** [primefaces/community#1601] [PrimeVue]
OverlayPanel is always placed underneath target element. We'd like a prop to specify whether to go on top, topleft, topright, right, righttop, rightbottom, bottom, bottomleft, bottomright, left, lefttop, or leftbottom.

Bootstrap-vue has that option for its equivalent Popover component:
https://bootstrap-vue.org/docs/components/popover#positioning

**Comment by JohnCampionJr** [comment]
This is a very missed feature.  Any chance we could get it soon?

**Comment by abhishek-junghare** [comment]
Yes, indeed. I want the popover to open at the top as default, if not then at the bottom.

**Comment by huchenlei** [comment]
+1

**Comment by mourad1081** [comment]
+1

**Comment by robertmylne** [comment]
How is this not a thing. This alone is enough reason not to use this UI framework. Otherwise the dropdowns in the right hand column of tables look weird af. The ability to position is there as it does it automatically relative to the viewport. Just allow us passing a prop to handle it.

**Comment by ZiadJ** [comment]
I ended up writing a wrapper component to provide the missing functionality:
https://stackblitz.com/edit/primevue-extended
https://github.com/ZiadJ/primevue-extended
